src/select/__tests__/analytics-metadata.test.tsx

```diff
@@ -0,0 +1,227 @@
+
+import buttonTriggerLabels from '../../../lib/components/internal/components/button-trigger/analytics-metadata/styles.css.js';
+import optionLabels from '../../../lib/components/internal/components/option/analytics-metadata/styles.css.js';
+import selectableItemsLabels from '../../../lib/components/internal/components/selectable-item/analytics-metadata/styles.css.js';
+
+const labels = { ...selectableItemsLabels, ...optionLabels, ...buttonTriggerLabels };
+
+function renderSelect(props: Partial<SelectProps>) {
+  const renderResult = render(
+    <Select options={options} selectedOption={null} ariaLabel="select with metadata" onChange={() => {}} {...props} />
+  );
+  return createWrapper(renderResult.container).findSelect()!;
+}
+
+const getMetadataContexts = (label = 'select with metadata', disabled?: boolean) => {
+  const metadata: GeneratedAnalyticsMetadataFragment = {
+    contexts: [
+      {
+        type: 'component',
+        detail: {
+          name: 'awsui.Select',
+          label,
+          properties: {
+            disabled: disabled ? 'true' : 'false',
+          },
+        },
+      },
+    ],
+  };
+  return metadata;
+};
+
+const options: SelectProps['options'] = [
+  { value: 'value1' },
+  { value: 'value2', label: 'label2' },
+  { value: 'value4', label: 'label4', disabled: true },
+];
+const optionsWithGroups: SelectProps['options'] = [
+  {
+    label: 'group1',
+    options: [{ value: 'value1', label: 'label1' }, { value: 'value1repeated' }],
+  },
+  {
+    label: 'group2',
+    disabled: true,
+    options: [{ value: 'value2', label: 'label2' }],
+  },
+  {
+    label: 'group3',
+    options: [{ value: 'value3', label: 'label3' }],
+  },
+  {
+    label: 'group4',
+    options: [
+      { value: 'value4', label: 'label4', disabled: true },
+      { value: 'value5', label: 'label5', disabled: true },
+    ],
+  },
+  {
+    label: 'group5',
+    options: [
+      { value: 'value6', label: 'label6' },
+      { value: 'value7', label: 'label7' },
+    ],
+  },
+];
+
+beforeAll(() => {
+  activateAnalyticsMetadata(true);
+});
+describe('Select renders correct analytics metadata', () => {
+  test('on the trigger button', () => {
+    const wrapper = renderSelect({});
+    let trigger = wrapper.findTrigger()!.getElement();
+    validateComponentNameAndLabels(trigger, labels);
+    expect(getGeneratedAnalyticsMetadata(trigger)).toEqual({
+      action: 'expand',
+      detail: {
+        label: 'select with metadata',
+        expanded: 'true',
+      },
+      ...getMetadataContexts(),
+    });
+
+    wrapper.openDropdown();
+    trigger = wrapper.findTrigger()!.getElement();
+    validateComponentNameAndLabels(trigger, labels);
+    expect(getGeneratedAnalyticsMetadata(trigger)).toEqual({
+      action: 'expand',
+      detail: {
+        label: 'select with metadata',
+        expanded: 'false',
+      },
+      ...getMetadataContexts(),
+    });
+  });
+  test('when disabled', () => {
```

Comment:
If including readonly then should add a test for it too

Reply:
Done